frontend/src/components/ManagePermissionsTable/utils.ts

```diff
@@ -0,0 +1,241 @@
+            `;
+        default:
+            return `This table shows all the users in ${appName} and their permissions levels`;
+    }
+};
+
+/**
+ * Generates the label for a grant button based on the authorization group.
+ */
+export const getGrantButtonLabel = (authGroup: AdminRoleAuthGroup): string => {
+    switch (authGroup) {
+        case "FAM_ADMIN":
+            return "Add application admin";
+        case "APP_ADMIN":
+            return "Add user permission";
+        case "DELEGATED_ADMIN":
+            return "Create delegated admin";
+        default:
+            return "";
+    }
+};
+
+/**
+ * Generates a list of headers based on the authorization group used for skeleton.
+ */
+export const getHeaders = (authGroup: AdminRoleAuthGroup): string[] => {
+    switch (authGroup) {
+        case "FAM_ADMIN":
+            return [
+                "User Name",
+                "Domain",
+                "Full Name",
+                "Email",
+                "Application",
+                "Environment",
+                "Role",
+                "Action",
+            ];
+        case "APP_ADMIN":
+            return [
+                "User Name",
+                "Domain",
+                "Full Name",
+                "Email",
+                "Role",
+                "Action",
+            ];
+        case "DELEGATED_ADMIN":
+            return [
+                "User Name",
+                "Domain",
+                "Full Name",
+                "Email",
+                "Role Enabled To Assign",
+                "Action",
+            ];
+        default:
+            return [];
+    }
+};
+
+// Allow customized message templates for success and error cases
+type NotificationContext<T> = {
+    action: string;
+    entityName: string;
+    successTemplate: (variables: T) => string;
+    errorTemplate: (variables: T, error: Error) => string;
+};
+
+// For deleteAppAdminMutation
+export const deleteAppAdminContext: NotificationContext<FamApplicationUserRoleAssignmentGetSchema> =
+    {
+        action: "remove",
+        entityName: "access",
+        successTemplate: (variables) => `
+        You removed ${
+            variables.role?.display_name || ""
+        } access from ${formatUserNameAndId(
+            variables.user.user_name,
+            variables.user.first_name,
+            variables.user.last_name
+        )}
+    `,
+        errorTemplate: (variables, error) => `
+        Failed to remove ${
+            variables.role?.display_name || ""
+        } access from ${formatUserNameAndId(
+            variables.user.user_name,
+            variables.user.first_name,
+            variables.user.last_name
+        )}. Error: ${
+            isAxiosError(error) ? formatAxiosError(error) : error.message
+        }
+    `,
+    };
+
+// For deleteDelegatedAdminMutation
+export const deleteDelegatedAdminContext: NotificationContext<FamAccessControlPrivilegeGetResponse> =
```

Comment:
Maybe same here for variable name, `deleteDelegatedAdminContext` look strange, just add `NotificationContext` at the end.

Reply:
https://github.com/bcgov/nr-forests-access-management/pull/1650/commits/87683359832be05ccaf651ffef906f03aa42e407
renamed all